File: app/src/main/java/eu/toma/dev/playground/warcraft/WoWBaseFragment.java
package eu.toma.dev.playground.warcraft;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;

import eu.toma.dev.playground.HasComponent;
import eu.toma.dev.playground.warcraft.di.WoWcomponent;

/**
* Base adapter for all WoW Fragments
*/
public class WoWBaseFragment extends Fragment implements HasComponent<WoWcomponent>
{
    @NonNull
    @Override
    public WoWcomponent getComponent()
    {
        return ((WarcraftActivity)getActivity()).getComponent();
    }
}

File: app/src/main/kotlin/warcraft/WoWBaseFragment.kt
package warcraft

import android.os.Bundle
import android.support.v4.app.Fragment

import eu.toma.dev.playground.HasComponent
import warcraft.di.WoWcomponent

/**
 * Base class for all WoW Fragments
 */
open class WoWBaseFragment : Fragment(), HasComponent<WoWcomponent> {
    override fun getComponent(): WoWcomponent {
        return (activity as WarcraftActivity).component
    }

    open fun onActivityCreatedDelegate(savedInstanceState: Bundle?){
        // Override in subclass
    }

    override final fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        // Perform cast to see if the fragment is attached to wrong activity
        activity as WarcraftActivity
        onActivityCreatedDelegate(savedInstanceState)
    }
}